### ISSUE TITLE
Debug is killed by an array with a freed node

### ISSUE THREAD

**Issue by ripperdave** [godotengine/godot#30936]
The debug is killed by this situation. The problem is that there is no error messages or warnings.
Imagine a massive project, you debug it and suddenly the debug stops.

**Godot version:**
3.1.1 official

**OS/device including version:**
Ubuntu 18.04.02 LTS.
Intel® Core™ i5-2500K CPU @ 3.30GHz × 4
GeForce GTX 1050/PCIe/SSE2
Gnome: 3.28.2
64-bit

**Issue description:**
Add a breakpoint to the DebugError.gd
Run step by step (press F10)
As the array has a node that is freed the debuging stops.

Probably the inspector tries to read and show the content of the array.

**Steps to reproduce:**
Run this project.
[ArrayInvalidNode.zip](https://github.com/godotengine/godot/files/3443982/ArrayInvalidNode.zip)

**Minimal reproduction project:**

**Comment by qarmin** [comment]
Backtrace
```
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x3ef20) [0x7f3f4e992f20] (??:0)
[2] List<Pair<Variant const*, Variant>, DefaultAllocator>::size() const (/home/rafal/Pulpit/godot/./core/list.h:410)
[3] OrderedHashMap<Variant, Variant, VariantHasher, VariantComparator, (unsigned char)3, (unsigned char)8>::size() const (/home/rafal/Pulpit/godot/./core/ordered_hash_map.h:275)
[4] Dictionary::empty() const (/home/rafal/Pulpit/godot/core/dictionary.cpp:130)
[5] Object::get_property_list(List<PropertyInfo, DefaultAllocator>*, bool) const (/home/rafal/Pulpit/godot/core/object.cpp:638)
[6] encode_variant(Variant const&, unsigned char*, int&, bool) (/home/rafal/Pulpit/godot/core/io/marshalls.cpp:1094)
[7] encode_variant(Variant const&, unsigned char*, int&, bool) (/home/rafal/Pulpit/godot/core/io/marshalls.cpp:1201)
[8] ScriptDebuggerRemote::_put_variable(String const&, Variant const&) (/home/rafal/Pulpit/godot/core/script_debugger_remote.cpp:111)
[9] ScriptDebuggerRemote::debug(ScriptLanguage*, bool) (/home/rafal/Pulpit/godot/core/script_debugger_remote.cpp:226)
[10] GDScriptLanguage::debug_break(String const&, bool) (/home/rafal/Pulpit/godot/modules/gdscript/gdscript_editor.cpp:237)
[11] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Variant::CallError&, GDScriptFunction::CallState*) (/home/rafal/Pulpit/godot/modules/gdscript/gdscript_function.cpp:1527 (discriminator 3))
[12] GDScriptInstance::_ml_call_reversed(GDScript*, StringName const&, Variant const**, int) (/home/rafal/Pulpit/godot/modules/gdscript/gdscript.cpp:1207)
[13] GDScriptInstance::call_multilevel_reversed(StringName const&, Variant const**, int) (/home/rafal/Pulpit/godot/modules/gdscript/gdscript.cpp:1216)
[14] Node::_notification(int) (/home/rafal/Pulpit/godot/scene/main/node.cpp:152)
[15] Node::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/main/node.h:46 (discriminator 14))
[16] CanvasItem::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/2d/canvas_item.h:166 (discriminator 3))
[17] Node2D::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/2d/node_2d.h:38 (discriminator 3))
[18] Object::notification(int, bool) (/home/rafal/Pulpit/godot/core/object.cpp:945)
[19] Node::_propagate_ready() (/home/rafal/Pulpit/godot/scene/main/node.cpp:197)
[20] Node::_propagate_ready() (/home/rafal/Pulpit/godot/scene/main/node.cpp:186 (discriminator 2))
[21] Node::_set_tree(SceneTree*) (/home/rafal/Pulpit/godot/scene/main/node.cpp:2573)
[22] SceneTree::init() (/home/rafal/Pulpit/godot/scene/main/scene_tree.cpp:458)
[23] OS_X11::run() (/home/rafal/Pulpit/godot/platform/x11/os_x11.cpp:3171)
[24] /usr/bin/godot(main+0x128) [0x1383b1f] (/home/rafal/Pulpit/godot/platform/x11/godot_x11.cpp:57)
[25] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7f3f4e975b97] (??:0)
[26] /usr/bin/godot(_start+0x2a) [0x138393a] (??:?)
```

**Comment by KoBeWi** [comment]
Fixed in 3.2
